assets/src/util/mapMode.tsx

```diff
@@ -9,7 +9,7 @@ export interface NavMode {
   path: string
   title: string
   element: ReactElement
-  navIcon: (props: any) => ReactElement
+  navIcon: React.JSXElementConstructor<any>
```

Comment:
**question (non-blocking):** I wonder if there's some type that could be filled in for the parameter to the generic more specific than `any` - I usually try to avoid uses of `any` since it basically means "this should typecheck with whatever you throw at it," but I don't have any ideas about what else to use.

Reply:
what about this?
```suggestion
  navIcon: React.JSXElementConstructor<ComponentPropsWithoutRef<"var">>
```

Reply:
I think this is a list of all the `JSX.IntrinsicElements` which use only the base attributes allowed on all HTML Elements?

```
assets/node_modules/@types/react/index.d.ts:
  3143:             abbr: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3144:             address: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3146:             article: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3147:             aside: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3149:             b: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3151:             bdi: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3152:             bdo: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3153:             big: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3159:             caption: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3160:             cite: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3161:             code: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3166:             dd: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3169:             dfn: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3173:             dt: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3174:             em: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3177:             figcaption: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3178:             figure: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3179:             footer: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3188:             header: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3189:             hgroup: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3192:             i: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3197:             kbd: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3203:             main: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3205:             mark: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3207:             menuitem: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3210:             nav: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3211:             noindex: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3212:             noscript: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3220:             picture: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3224:             rp: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3225:             rt: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3226:             ruby: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3227:             s: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3228:             samp: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3231:             section: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3233:             small: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3236:             strong: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3238:             sub: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3239:             summary: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3240:             sup: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3253:             u: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3255:             "var": React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
  3257:             wbr: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
```

Reply:
@mbta/skate-developers what do you think of this?
```suggestion
  navIcon: React.JSXElementConstructor<React.PropsWithoutRef<React.HTMLAttributes<HTMLElement>>>
```

Reply:
Sounds good to me if it typechecks!

Reply:
see: 14e4c28cd7db394837a0ac7f1289de3f22883c16